scripts/unsupported/reheader_bam/reheader_bam.py

```diff
@@ -0,0 +1,96 @@
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+from copy import deepcopy
+import os
+import subprocess
+import pysam
+import random
+
+# DO NOT SEED
+#random.seed("3337")
+
+def parse_options(description, epilog):
+    parser = ArgumentParser(description=description, formatter_class=RawDescriptionHelpFormatter, epilog=epilog)
+    parser.add_argument('input_file', type=str, help='')
+    parser.add_argument('contigs', type=str, help='The contigs to PRESERVE separated by "," ... THESE MUST BE SORTED (e.g. 5,20,21 is okay ; 20,5,21 is not okay)')
+    parser.add_argument('output_file', type=str, help='')
+
+    # Process arguments
+    args = parser.parse_args()
+
+    return args
+
+
+def main():
+    args = parse_options("Tool to reheader a (copy of a) bam to only the specified contigs.  EXPERIMENTAL and UNSUPPORTED", """WARNING:  No error checking.
+    Assumes that the bam is coordinate sorted and paired-end.
+    java -jar picard.jar ReplaceSamHeader HEADER=tmp_header.sam I=<(cat <(head -n1  tmp_header.sam ) <(samtools view tumor_1_foo.bam)) O=yossi.bam CREATE_INDEX=true VALIDATION_STRINGENCY=SILENT
+
+    THIS SCRIPT CAN BE VERY SLOW ON LARGE BAMS
+
+    This script is mostly meant for generating test bams that validate (even in Picard).
+
+    Example: python reheader_bam.py some.bam 20,21 some_20_21_only.bam""")
+
+    input_file = args.input_file
+    output_file = args.output_file
+    contigs = args.contigs.split(",")
+    contigs = set([c.strip() for c in contigs])
+
+    # 1) BAM -> Filtered SAM w/ old header
+    random_num = random.random()
+    samfile_out_filename_step1 = os.path.dirname(os.path.abspath(output_file)) + "/tmp_step_1_" + str(random_num) + ".sam"
+
+    bamfile_in = pysam.AlignmentFile(input_file, 'rb')
+
+    samfile_out = pysam.AlignmentFile(samfile_out_filename_step1, 'wh', header=bamfile_in.header)
+    for i,r in enumerate(bamfile_in):
+        if r.next_reference_name in contigs:
+            try:
+                s_mate = bamfile_in.mate(r)
+                samfile_out.write(r)
+            except ValueError as ve:
+                if ve.message.find("fetch called") != -1 or ve.message.find("fetching by") != -1:
+                    raise ve
+
+        if i % 100 == 0:
```

Comment:
1000 makes you worry that it froze, though....

No action unless you really want it.